0004/README.md

```diff
@@ -0,0 +1,514 @@
+
+### From alerting-only to enforcing
+
+We expect that initially the policy evaluator will alert for an unexpected
+change, and not block it.
+
+-  An auditing policy can be evaluated post-upload and post-installation.
+    This minimizes interference with existing npm tooling. 
+-  An enforcing policy needs to be more tightly integrated with npm tools,
+    so that it can block a non-compliant package.
+
+Future work is to integrate the policy evaluator with npm tooling.
+
+### A proposed TOFU policy for a single npm package
+
+This is for a policy that lists explicit expectations for provenance values,
+instead of expectations for what provenance fields are invariant (regardless of
+their value).
+
+Inputs: 
+
+-  Package name and version.
+-  Optional: verified package provenance.
+-  Preferences. Examples include, but are not limited to:
+    -  Names of ‘sensitive' provenance properties that are not expected
+        to change such as builder ID or source repository,
+    -  Minimum SLSA level for an acceptable artifact, 
+    -  Maximum SLSA level depending on the builder. 
+
+-  Preferences may differ for different evaluator contexts: developer,
+    registry, or end user, and for a private or public registry.
+
+If no policy exists for this package, generate one with: 
+
+-  Package name 
+-  Version pattern that matches this package version and onwards.
+-  Expected provenance properties (SLSA level, signer, builder, source repo,
+    etc.).
+
+If a policy already exists for this package:
+
+-  Flag missing or unverifiable provenance.
+-  Flag an unexpected change in a sensitive provenance property (SLSA level,
+    source repo, etc.).
+
+The following will require approval:
+
+-  If a ‘good' package version does not satisfy expectations, add new
+    expectations, qualified with a version pattern that matches this package
+    version and onwards.
+
+## Main use cases
+
+### Use case 1. Publishing a new package version
+
+####  Existing process
+
+-  A developer runs "npm publish" to submit a new version of their npm
+    package to a registry.
+
+#### New behavior
+
+-  The package registry does a policy check for the package provenance,
+    initially non-enforcing.
+-  A private package registry may impose different expectations than a
+    public one. 
+
+#### New process
+
+-  A developer may also do a policy check in preparation for package upload.
+
+#### What packages will be covered by policy
+
+-  The policy applies only to the package that is being published (the
+    concrete versions of dependencies will be determined when a package is
+    downloaded for installation).
+
+####  Existing process details
+
+-  Besides package-specific code, an npm package contains a metadata file
+    (package.json) with the name and version of the package itself, and
+    optionally the names and allowed version ranges of direct dependencies.
+-  See "[Npm background](#heading=h.napkqzk41aim)" for more.
```

Comment:
I don't think this background info section was copied over from the original Google doc?

Reply:
Sorry that should be a link to the "npm background"  section  in *this* document. The link was not updated when the Google doc was converted to markdown. I'll update the README.md.

Reply:
I updated the quotes as suggested and replaced all in-document links.